create_auction_result.php

```diff
@@ -1,33 +1,121 @@
 <?php include_once("header.php")?>
+<?php include_once("create_auction.php")?>
+<?php include_once("database.php"); ?>
 
 <div class="container my-5">
 
 <?php
 
-// This function takes the form data and adds the new auction to the database.
+    session_start();
+    global $connection;
+    if (!isset($_SESSION) || $_SESSION == null) {
+    echo('<div class="text-center">You\'re not logged in. Please re-login if this was a mistake</div>');
+    header('refresh:3;url=browse.php');
+    }
 
-/* TODO #1: Connect to MySQL database (perhaps by requiring a file that
-            already does this). */
+    // This function takes the form data and adds the new auction to the database.
 
+    // $mysqli = connectToDatabase();
 
-/* TODO #2: Extract form data into variables. Because the form was a 'post'
-            form, its data can be accessed via $POST['auctionTitle'], 
-            $POST['auctionDetails'], etc. Perform checking on the data to
-            make sure it can be inserted into the database. If there is an
-            issue, give some semi-helpful feedback to user. */
+    // Check if the form was submitted
+    if (isset($_POST["auctionsubmit"])) {
+   
+    // Extract form data into variables. 
+    $title = $_POST['auctionTitle'];
+    $details = $_POST['auctionDetails'];
+    $startPrice = $_POST['auctionStartPrice'];
+    $reservePrice = $_POST['auctionReservePrice'];
+    $endDate = $_POST['auctionEndDate'];
+    $formattedEndDate = date('Y-m-d H:i:s', strtotime($endDate));
 
+    // print_r($_SESSION);
+    $seller_id = $_SESSION['id'];
 
-/* TODO #3: If everything looks good, make the appropriate call to insert
-            data into the database. */
-            
+    // TODO #2.1: Perform data validation.
+    $errors = [];
 
-// If all is successful, let user know.
-echo('<div class="text-center">Auction successfully created! <a href="FIXME">View your new listing.</a></div>');
+    // Check if title is not empty
+    if (empty($title)) {
+        $errors[] = "Title cannot be empty.";
+    }
 
+    // Check if details are not empty
+    // if (empty($details)) {
+    //     $errors[] = "Details cannot be empty.";
+    // }
+
+    //  Validate category - you may want to check if the selected category exists in the database
+    // if (!is_numeric($category) || $category <= 0) {
+    //     $errors[] = "Invalid category.";
+    // }
+
+    // Check if start price is a positive number
+    if (!is_numeric($startPrice) || $startPrice <= 0) {
+        $errors[] = "Start price must be a positive number.";
+    }
+
+    // Check if reserve price is a positive number (if provided)
+    if (!isset($reservePrice) || trim($reservePrice) === '') {
+        // If reserve price is blank, assign the start price to it
+        $reservePrice = $startPrice;
+    } elseif (!is_numeric($reservePrice) || $reservePrice <= 0) {

```

Comment:
Should we also not have a check for the reserve price to be greater than the start price?

I might be fuzzy on the logic of having a reserve price, so pls correct me If I'm wrong @jasmehr @gazalguptaa @paulinalydav

Reply:
Yeah, and need this logic too on the crob job later